core/types/receipt.go

```diff
@@ -241,3 +294,12 @@ func (r Receipts) GetRlp(i int) []byte {
 	}
 	return bytes
 }
+
+// AreMissingMetadata returns whether or not the receipts are missing their
+// metadata fields.
+func (r Receipts) AreMissingMetadata() bool {
+	// The transaction hash was removed at the same time as GasCost, which
+	// is hard to detect given that it's a uint64 (on mainnet it is impossible,
+	// but could be the case on a private chain that changed gas costs).
```

Comment:
This method can go away with the refactoring. Will remove. This is a typo, I meant `GasUsed` instead of `GasCost`. All transactions will at least use at`21000` gas so `GasUsed` should never be 0 is my understanding (even if the whole transaction processing cost is refunded).